apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -338,7 +339,7 @@ export default class ProjectManager {
 
   setThumbnail(pngBlob: Blob) {
     this.thumbnailPngBlob = pngBlob;
-    this.thumbnailUrl = `/v3/files/${this.channelId}/.metadata/thumbnail.png`;
+    this.thumbnailUrl = getProjectThumbnailUrl(this.channelId);
```

Comment:
Is this needed anymore?

Reply:
Refer to `this.thumbnailUrl` below to decide whether we want to save to channel.
```
      if (this.thumbnailUrl && !this.lastChannel?.thumbnailUrl) {
        this.channelToSave = {
          ...this.channelToSave,
          thumbnailUrl: this.thumbnailUrl,
        };
      }
```

Reply:
Could we use the existence of `thumbnailPngBlob` instead?

Reply:
I considered that, but `thumbnailPngBlob` is set to `undefined`after `saveThumbnail` is called. Could set it to `undefined` after. I kinda like setting it after saving thumbnail. But don't have a strong preference.
```
      if (this.thumbnailPngBlob && !this.lastChannel?.thumbnailUrl) {
        this.channelToSave = {
          ...this.channelToSave,
          thumbnailUrl: getProjectThumbnailUrl() 
        };
      }
      this.thumbnailPngBlob = undefined;
```

Reply:
Yes, we save URL to channel and access it when displaying thumbnail in the personal/public galleries. Thanks!